src/index.ts

```diff
@@ -134,11 +138,16 @@ function requestToFetchOptions(reqOpts: Options) {
     process.env.HTTP_PROXY ||
     process.env.http_proxy ||
     process.env.HTTPS_PROXY ||
-    process.env.https_proxy;
-  if (reqOpts.proxy || proxy) {
-    options.agent = new HttpsProxyAgent(proxy);
+    process.env.https_proxy ||
+    reqOpts.proxy;
+  if (proxy) {
+    options.agent = /^http:\/\//.test(uri)
```

Comment:
Should we cache this outside of the checks to avoid duplication? Or maybe even just cache the regex if we're trying to avoid testing unnecessarily